app/main.py

```diff
@@ -1,3 +1,57 @@
+from __future__ import annotations
+
+
 class Distance:
-    # Write your code here
-    pass
+    def __init__(self, km: float | int) -> None:
+        self.km = km
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __add__(self, other: Distance | int | float) -> Distance:
+        if isinstance(other, (int, float)):
+            return Distance(self.km + other)
+        return Distance(self.km + other.km)
+
+    def __iadd__(self, other: Distance | float | int) -> Distance:
+        if isinstance(other, (int, float)):
+            self.km += other
+        else:
+            self.km += other.km
+        return self
+
+    def __mul__(self, other: float | int) -> Distance:
+        return Distance(self.km * other)
+
+    def __truediv__(self, other: int | float) -> Distance:
+        if other == 0:
+            print("Error dividing by zero!")
+        return Distance(round(self.km / other, 2))
```

Comment:
In the `__truediv__` method, instead of printing an error message when dividing by zero, you should raise an exception, such as `ZeroDivisionError`, to properly handle this error case. This will allow the calling code to catch and handle the exception appropriately.